tests/unit/dataactvalidator/test_sf133_content.py

```diff
@@ -0,0 +1,161 @@
+                '1029', '1030', '1031', '1032', '1033', '1034', '1035', '1036', '1037', '1038', 
+                '1039', '1040', '1041', '1042', '1160', '1180', '1260', '1280', '1340', '1440', 
+                '1540', '1640', '1750', '1850', '1910', '2190', '2490', '2500'
+            ]
+            data = data[(data.line.isin(sf_133_validation_lines) & (data.amount!=0))]
+
+            # create list of unique TAS
+            data.sort_values(by=['tas'], inplace=True)
+            data.set_index(keys=['tas'], drop=False, inplace=True)
+            all_tas=data['tas'].unique().tolist()
+
+            # iterate through TAS list
+            for tas in all_tas:
+                current_tas = data.loc[data.tas==tas]
+
+                sum_one = 0
+                validate_one = 0
+                validate_one_lines = [
+                    '1000', '1010', '1011', '1012', '1013', '1014', '1015', '1016', '1017', 
+                    '1018', '1019', '1020', '1021', '1022', '1023', '1024', '1025', '1026', 
+                    '1027', '1028', '1029', '1030', '1031', '1032', '1033', '1034', '1035', 
+                    '1036', '1037', '1038', '1039', '1040', '1041', '1042', '1160', '1180', 
+                    '1260', '1280', '1340', '1440', '1540', '1640', '1750', '1850'
+                ]
+
+                sum_two = 0
+                validate_two = 0
+                validate_two_lines = ['2190', '2490']
+                for index, row in current_tas.iterrows():
+                    if row.line in validate_one_lines:
+                        sum_one += row.amount
+                    elif row.line=='1910':
+                        validate_one = float("{0:.2f}".format(row.amount))
+                    elif row.line in validate_two_lines:
+                        sum_two += row.amount
+                    elif row.line=='2500':
+                        validate_two = float("{0:.2f}".format(row.amount))
+                sum_one = float("{0:.2f}".format(sum_one))
+                sum_two = float("{0:.2f}".format(sum_two))
+
+                validation_one = sum_one==validate_one
+                validation_two = sum_two==validate_two
+                validation_three = validate_one==validate_two
+
+                tas_m = TAS_RE.match(tas)
+
+                # initial validations fail
+                if not (validation_one and validation_two) and (tas_m and tas_m.group('atc')!='X' and file_fiscal_year>int(tas_m.group('epoa'))):
+                    if not validation_three:
+                        failed_validations.append('{0},{1},{2},{3},{4},{5},{6},{7},1900!=2500,{8},{9}'.format(
+                            file_index, tas_m.group('ata'), tas_m.group('aid'), 
+                            tas_m.group('atc'), tas_m.group('bpoa'), tas_m.group('epoa'), 
+                            tas_m.group('mac'), tas_m.group('sac'), validate_one, validate_two)
+                        )
```

Comment:
I switched back and forth when deciding which one would be easier to understand. I'm not sure why I decided on `format()` over `join()`, because looking over it again I agree that `join()` is more readable.